### PR TITLE
Np 47010 resource service assume eventual consistency will happen

### PR DESCRIPTION
**Pull request by anetteOlli**
Ikke helt trivielt å bytte ut da det ser ut til at roundtrippen via dtoMapper gir muterbare collections, mens vi setter alle tingene våre via unmutable collections.

### CODE REVIEW

File: brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java [hunk 1]
```diff
@@ -1932,22 +1932,6 @@ private String getPartOfReport(S3Event s3Event, NvaBrageMigrationDataGenerator b
         return s3Driver.getFile(uri.toS3bucketPath());
     }
 
-    private Publication persistPublicationWithHandleInAdditionalIdentifiers(String cristinIdentifier, String year,
```
Comment (anetteOlli) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java:1935]: removed unused method

File: brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java [hunk 2]
```diff
@@ -953,8 +954,7 @@ void shouldInjectBrageHandleAsAdditionalIdentifierOnlyWhenMergingDegreeWithPubli
 
 
         assertThat(handles, hasItem(expectedAdditionalIdentifier));
-
-        var set = existingPublication.getAdditionalIdentifiers();
```
Comment (anetteOlli) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java:957]: apparently roundtripping a publication via Jackson returns a mutable set.

File: brage-import/src/test/java/no/sikt/nva/brage/migration/testutils/ReferenceGenerator.java [hunk 3]
```diff
@@ -293,7 +294,7 @@ private static Reference buildReference(Builder builder) {
             }
             if (NvaType.VISUAL_ARTS.getValue().equals(builder.getType().getNva())) {
                 return new Reference.Builder().withPublicationInstance(
-                        new VisualArts(VisualArtsSubtype.createOther(null), null, null))
```
Comment (anetteOlli) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/testutils/ReferenceGenerator.java:296]: round tripping via Jackson gives you null value, but not you get empty sets or list

File: publication-commons/src/main/java/no/unit/nva/publication/exception/DynamoDBException.java [hunk 4]
```diff
@@ -1,20 +0,0 @@
-package no.unit.nva.publication.exception;
-
-import static org.apache.http.HttpStatus.SC_BAD_GATEWAY;
-import nva.commons.apigateway.exceptions.ApiGatewayException;
-
-public class DynamoDBException extends ApiGatewayException {
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/exception/DynamoDBException.java:6]: unused class except for inside tests

File: publication-commons/src/main/java/no/unit/nva/publication/exception/ErrorResponseException.java [hunk 5]
```diff
@@ -1,16 +0,0 @@
-package no.unit.nva.publication.exception;
-
-import nva.commons.apigateway.exceptions.ApiGatewayException;
-import org.apache.http.HttpStatus;
-
-public class ErrorResponseException extends ApiGatewayException {
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/exception/ErrorResponseException.java:6]: unused class except for inside tests

File: publication-commons/src/main/java/no/unit/nva/publication/exception/InternalErrorException.java [hunk 6]
```diff
@@ -1,16 +0,0 @@
-package no.unit.nva.publication.exception;
-
-import java.net.HttpURLConnection;
-import nva.commons.apigateway.exceptions.ApiGatewayException;
-
-public class InternalErrorException extends ApiGatewayException {
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/exception/InternalErrorException.java:6]: unused class except for inside tests

File: publication-commons/src/main/java/no/unit/nva/publication/exception/NoResponseException.java [hunk 7]
```diff
@@ -1,16 +0,0 @@
-package no.unit.nva.publication.exception;
-
-import nva.commons.apigateway.exceptions.ApiGatewayException;
-import org.apache.http.HttpStatus;
-
-public class NoResponseException extends ApiGatewayException {
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/exception/NoResponseException.java:6]: unused class except for inside tests

File: publication-commons/src/main/java/no/unit/nva/publication/exception/NotImplementedException.java [hunk 8]
```diff
@@ -1,18 +0,0 @@
-package no.unit.nva.publication.exception;
-
-import nva.commons.apigateway.exceptions.ApiGatewayException;
-import org.apache.http.HttpStatus;
-
-public class NotImplementedException extends ApiGatewayException {
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/exception/NotImplementedException.java:6]: unused class except for inside tests

File: publication-commons/src/main/java/no/unit/nva/publication/model/storage/DynamoEntry.java [hunk 9]
```diff
@@ -21,7 +21,6 @@
 public interface DynamoEntry {
 
     String CONTAINED_DATA_FIELD_NAME = "data";
-    boolean NO_WRAP = true;
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/model/storage/DynamoEntry.java:24]: unused constant

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 10]
```diff
@@ -72,8 +72,6 @@ public class ResourceService extends ServiceWithTransactions {
     public static final Supplier<SortableIdentifier> DEFAULT_IDENTIFIER_SUPPLIER = SortableIdentifier::next;
     public static final int AWAIT_TIME_BEFORE_FETCH_RETRY = 50;
     public static final String RESOURCE_REFRESHED_MESSAGE = "Resource has been refreshed successfully: {}";
-    public static final String INVALID_PATH_ERROR = "The document path provided in the update expression is invalid "
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:75]: unused constant

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ServiceWithTransactions.java [hunk 11]
```diff
@@ -32,11 +31,7 @@
 public class ServiceWithTransactions {
 
     public static final String EMPTY_STRING = "";
-    public static final String DOUBLE_QUOTES = "\"";
```
Comment (anetteOlli) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ServiceWithTransactions.java:35]: unused constants

File: publication-commons/src/test/java/no/unit/nva/publication/exception/DynamoDBExceptionTest.java [hunk 12]
```diff
@@ -1,16 +0,0 @@
-package no.unit.nva.publication.exception;
-
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.core.Is.is;
-import static org.hamcrest.core.IsEqual.equalTo;
-import java.net.HttpURLConnection;
-import org.junit.jupiter.api.Test;
-
-public class DynamoDBExceptionTest {
```
Comment (anetteOlli) [inline publication-commons/src/test/java/no/unit/nva/publication/exception/DynamoDBExceptionTest.java:9]: Unused exception

File: publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java [hunk 13]
```diff
@@ -287,16 +286,6 @@ void createResourceSavesResourcesWithSameOwnerAndPublisherButDifferentIdentifier
         assertDoesNotThrow(() -> createPersistedPublicationWithDoi(resourceService, anotherResource));
     }
 
-    @Test
-    void createPublicationReturnsNullWhenResourceDoesNotBecomeAvailable() throws BadRequestException {
```
Comment (anetteOlli) [inline publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java:291]: Test no longer applicable.

**Comment by anetteOlli** [comment]
Så forsvant ikke polling metoden 100 %. TicketService kaster exception når den ikke oppnår eventual consistency selv, så jeg lurer på om jeg skal endre der eller ikke. Men kan uansett starte med dette.

File: brage-import/src/test/java/no/sikt/nva/brage/migration/testutils/ReferenceGenerator.java [hunk 14]
```diff
@@ -293,7 +294,7 @@ private static Reference buildReference(Builder builder) {
...
+                        new VisualArts(VisualArtsSubtype.createOther(null), null, Set.of()))
```
Comment (brinxmat) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/testutils/ReferenceGenerator.java:297]: I'd prefer `Collections.emptySet()` as it looks deliberate

File: brage-import/src/test/java/no/sikt/nva/brage/migration/testutils/ReferenceGenerator.java [hunk 15]
```diff
@@ -326,7 +327,7 @@ private static Reference buildReference(Builder builder) {
             }
             if (NvaType.LITERARY_ARTS.getValue().equals(builder.getType().getNva())) {
                 return new Reference.Builder().withPublicationInstance(
-                    new LiteraryArts(LiteraryArtsSubtypeOther.createOther(null), null, null))
+                    new LiteraryArts(LiteraryArtsSubtypeOther.createOther(null), List.of(), null))
```
Comment (brinxmat) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/testutils/ReferenceGenerator.java:330]: Similarly, `Collections.emptyList()`

**Review by brinxmat — CHANGES_REQUESTED** [review]
A couple of minor changes, nice job

**Review by brinxmat — APPROVED** [review]

**Pull request merged**